src/schism/schism_esmf_util.F90

```diff
@@ -458,6 +458,10 @@ end subroutine SCHISM_InitializePtrMap
 #define ESMF_METHOD "SCHISM_FieldPtrUpdate"
 subroutine SCHISM_FieldPtrUpdate(field, farray, kwe, isPtr, rc)
 
+  use schism_glbl, only: ne, neg, nea
```

Comment:
Is it wise to use schism_glbl vars? Or could we better use the internalState wrapped structure,also for ne, neg, nea?  I see here the (partial) conversion from `isPtr%numOwnedNodes` to `np`.  Is that correctly implemented in all places?

Reply:
I think there is no harm to use modules from the model. We are always doing it. The internal state is just for storing some information in the cap.